__tests__/accountAbstraction.test.ts

```diff
@@ -0,0 +1,461 @@
+        5,
+        6
+      );
+      const tr1 = await provider.waitForTransaction(th1);
+      expect(tr1).toHaveProperty('execution_status', 'SUCCEEDED');
+
+      const call2 = contractETH.populate('transfer', {
+        recipient: account0.address,
+        amount: cairo.uint256(300_000),
+      });
+      const { transaction_hash: th2 } = await accountAbstraction.execute(
+        call2,
+        undefined,
+        undefined,
+        4, // inputs for abstraction of transaction are here
+        5,
+        6
+      );
+      const tr2 = await provider.waitForTransaction(th2);
+      expect(tr2).toHaveProperty('execution_status', 'SUCCEEDED');
+    });
+
+    test('Abstracted declare then deploy', async () => {
+      // declare
+      const declareResponse = await accountAbstraction.declare(
+        { contract: compiledDummy1Sierra, casm: compiledDummy1Casm },
+        undefined,
+        7, // inputs for abstraction of declare are here
+        8,
+        9
+      );
+      const classHashDummy1 = declareResponse.class_hash;
+      const tr3 = await provider.waitForTransaction(declareResponse.transaction_hash);
+      expect(tr3).toHaveProperty('execution_status', 'SUCCEEDED');
+
+      // deploy
+      const respDeploy = await accountAbstraction.deploy(
+        {
+          classHash: classHashDummy1,
+        },
+        undefined,
+        4, // inputs for abstraction of deploy are here.
+        5, // deploy is handled as a transaction.
+        6
+      );
+      const tr4 = await provider.waitForTransaction(respDeploy.transaction_hash);
+      expect(tr4).toHaveProperty('execution_status', 'SUCCEEDED');
+    });
+
+    test('Abstracted declareAndDeploy', async () => {
+      // declareAndDeploy
+      const response = await accountAbstraction.declareAndDeploy(
+        { contract: compiledDummy2Sierra, casm: compiledDummy2Casm },
+        undefined,
+        [7, 8, 9], // abstraction for declare
+        [4, 5, 6] //  abstraction for deploy
+      );
+      const tr4 = await provider.waitForTransaction(response.deploy.transaction_hash);
+      expect(tr4).toHaveProperty('execution_status', 'SUCCEEDED');
+    });
+
+    test('Abstracted messages', async () => {
+      const typedDataValidate: TypedData = {
+        domain: {
+          chainId: 'Starknet Mainnet',
+          name: 'Dappland',
+          version: '1.0',
+        },
+        message: {
+          MessageId: 345,
+          From: {
+            Name: 'Edmund',
+            Address: '0x7e00d496e324876bbc8531f2d9a82bf154d1a04a50218ee74cdd372f75a551a',
+          },
+          To: {
+            Name: 'Alice',
+            Address: '0x69b49c2cc8b16e80e86bfc5b0614a59aa8c9b601569c7b80dde04d3f3151b79',
+          },
+          Nft_to_transfer: {
+            Collection: 'Stupid monkeys',
+            Address: '0x69b49c2cc8b16e80e86bfc5b0614a59aa8c9b601569c7b80dde04d3f3151b79',
+            Nft_id: 112,
+            Negociated_for: {
```

Comment:
```suggestion
            Negotiated_for: {
```
is this property ours or? because it has a typo

Reply:
Our.
Will be modified.